.github/workflows/build_windows_pytorch_wheels.yml

```
@@ -53,30 +79,60 @@ jobs:
         with:
           python-version: ${{ inputs.python_version }}
 
+      # TODO(amd-justchen): share with build_windows_packages.yml. Include in VM image? Dockerfile?
+      - name: Install requirements
+        run: |
+          # ninja pinned due to a bug in the 1.13.0 release:
+          # https://github.com/ninja-build/ninja/issues/2616
+          choco install --no-progress -y ninja --version 1.12.1
+          choco install --no-progress -y awscli
+          echo "$PATH;C:\Program Files\Amazon\AWSCLIV2" >> $GITHUB_PATH
+
+      # After other installs, so MSVC get priority in the PATH.
+      - name: Configure MSVC
+        uses: ilammy/msvc-dev-cmd@0b201ec74fa43914dc39ae48a89fd1d8cb592756 # v1.13.0
+
       - name: Checkout PyTorch Source Repos
         run: |
-          python ./external-builds/pytorch/pytorch_torch_repo.py checkout
-      # TODO(scotttodd): include these
+          git config --global core.longpaths true
+          python ./external-builds/pytorch/pytorch_torch_repo.py checkout --repo ${{ env.CHECKOUT_ROOT }}/torch
+          # TODO(#910): Support torchvision and torchaudio on Windows
           # python ./external-builds/pytorch/pytorch_audio_repo.py checkout
           # python ./external-builds/pytorch/pytorch_vision_repo.py checkout
 
-      # TODO(scotttodd): build wheels, put into a directory location for next step to use
+      - name: Determine optional arguments passed to `build_prod_wheels.py`
+        if: ${{ inputs.rocm_version }}
+        run: |
+          pip install packaging
+          python build_tools/github_actions/determine_version.py \
+            --rocm-version ${{ inputs.rocm_version }}
 
-      # TODO(scotttodd): add upload once working
+      - name: Build PyTorch Wheels
+        shell: cmd
+        run: |
+          echo "Building PyTorch wheels for ${{ inputs.amdgpu_family }}"
+          python ./external-builds/pytorch/build_prod_wheels.py build --install-rocm --index-url "https://${{ inputs.cloudfront_url }}/${{ inputs.amdgpu_family }}/" --pytorch-dir ${{ env.CHECKOUT_ROOT }}/torch --clean --output-dir ${{ env.PACKAGE_DIST_DIR }} ${{ env.optional_build_prod_arguments }}
```

```suggestion
          python ./external-builds/pytorch/build_prod_wheels.py \
            build \
            --install-rocm \
            --index-url "https://${{ inputs.cloudfront_url }}/${{ inputs.amdgpu_family }}/" \
            --pytorch-dir ${{ env.CHECKOUT_ROOT }}/torch \
            --clean \
            --output-dir ${{ env.PACKAGE_DIST_DIR }} ${{ env.optional_build_prod_arguments }}
```
This would follow the style in `.github/workflows/build_linux_pytorch_wheels.yml` and might be easier to read.

---

Yeah, I see... you _could_ have dropped `shell: cmd` here to make it work but I see that this makes copying over and testing locally unnecessary more difficult. Whatever works best for you :)

---

Using bash here would allow the `\` characters to split the command across multiple lines, but the build process fails under bash (on my local machine and CI: https://github.com/ROCm/TheRock/issues/827#issuecomment-3025858800). That should be fixable, but it's a battle for another day.

It might be easier to get the rest of the workflow using cmd.

---

I'll try with `^` and add some breadcrumb comments where the shell is load bearing